Core/NPCBarrierGlow.cs

```diff
@@ -0,0 +1,164 @@
+﻿using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using Terraria.Graphics.Effects;
+using System;
+using Terraria;
+using Terraria.ModLoader;
+using StarlightRiver.Content.CustomHooks;
+
+namespace StarlightRiver.Core
+{ 
+    public class NPCBarrierGlowSystem : ModSystem
+    {
+        public override void PreUpdateNPCs()
+        {
+            NPCBarrierGlow.anyEnemiesWithBarrier = false;
+        }
+    }
+    public class NPCBarrierGlow : IOrderedLoadable
+    {
+        public static bool anyEnemiesWithBarrier = false;
+
+        public static RenderTarget2D NPCTarget;
+
+        public static RenderTarget2D NPCTargetBehindTiles;
+
+        public static Vector2 oldScreenPos = Vector2.Zero;
+
+        private static Color barrierColor => Color.Cyan;
+
+        public float Priority => 1.1f;
+
+        public void Load()
+        {
+            if (Main.dedServ)
+                return;
+
+            ResizeTarget();
+
+            Main.OnPreDraw += Main_OnPreDraw;
+            On.Terraria.Main.DrawNPCs += DrawBarrierOverlay;
+        }
+
+        public void Unload()
+        {
+            Main.OnPreDraw -= Main_OnPreDraw;
+            On.Terraria.Main.DrawNPCs -= DrawBarrierOverlay;
+        }
+
+        public static void ResizeTarget()
+        {
+            Main.QueueMainThreadAction(() =>
+            {
+                NPCTargetBehindTiles = new RenderTarget2D(Main.graphics.GraphicsDevice, Main.screenWidth, Main.screenHeight);
+                NPCTarget = new RenderTarget2D(Main.graphics.GraphicsDevice, Main.screenWidth, Main.screenHeight);
+            });
+        }
+
+        private void Main_OnPreDraw(GameTime obj)
+        {
+            GraphicsDevice gD = Main.graphics.GraphicsDevice;
```

Comment:
Whats with the strange capitilzation here? either go full uppercase or full lowercase for abbreviations